src/pages/Pool/PositionPage.tsx

```diff
@@ -346,7 +346,32 @@ const useInverter = ({
   }
 }
 
-export function PositionPage() {
+const INVALID_STATE = (
+  <PageWrapper>
+    <div style={{ display: 'flex', alignItems: 'center', flexDirection: 'column' }}>
+      <ThemedText.HeadlineLarge style={{ marginBottom: '8px' }}>
+        <Trans>Position unavailable</Trans>
+      </ThemedText.HeadlineLarge>
+      <ThemedText.BodyPrimary style={{ marginBottom: '32px' }}>
+        <Trans>To view a position, you must be connected to the network it belongs to.</Trans>
+      </ThemedText.BodyPrimary>
+      <PositionPageButtonPrimary as={Link} to="/pools" width="fit-content">
+        <Trans>Back to Pools</Trans>
+      </PositionPageButtonPrimary>
+    </div>
+  </PageWrapper>
+)
+
+export default function PositionPageWrapper() {
+  const { chainId } = useWeb3React()
+  if (isSupportedChain(chainId)) {
```

Comment:
Is it necessary to pull this into a separate component because the logic already in `PositionPage` will crash?

Reply:
yes exactly. without doing this the hooks in that component will run which we want to short circuit out of